### PR TITLE
Add support for TLS 1.3 ciphers.

### PR DESCRIPTION
**Pull request by pjbgf**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Adds support for TLS 1.3 ciphers as recommended by the [security audit](https://github.com/kubernetes/kubernetes/issues/81145). 

**Which issue(s) this PR fixes**:

Part of #81145

**Special notes for your reviewer**:
This change expands the list of possible TLS Ciphers that can be used within Kubernetes by adding 3 recommended ciphers for TLS 1.3, namely: 
```
TLS_AES_128_GCM_SHA256
TLS_CHACHA20_POLY1305_SHA256
TLS_AES_256_GCM_SHA384
```

**Does this PR introduce a user-facing change?**:

```release-note
Add support for TLS 1.3 ciphers: TLS_AES_128_GCM_SHA256, TLS_CHACHA20_POLY1305_SHA256 and TLS_AES_256_GCM_SHA384.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

/sig api-machinery
/sig auth
/area security
/priority backlog

### CODE REVIEW

**Comment by pjbgf** [comment]
@cji I am adding you here as you created the issue. 

/assign cji

**Comment by neolit123** [comment]
@kubernetes/sig-auth-pr-reviews

**Comment by liggitt** [comment]
hmm, I thought there was a unit test ensuring all supported ciphers were available in this map already… I would have expected a unit test failure if that was not the case

File: staging/src/k8s.io/component-base/cli/flag/ciphersuites_flag_test.go [hunk 1]
```diff
@@ -86,7 +86,8 @@ func TestConstantMaps(t *testing.T) {
 		if strings.HasPrefix(declName, "VersionTLS") {
 			discoveredVersions[declName] = true
 		}
-		if strings.HasPrefix(declName, "TLS_RSA_") || strings.HasPrefix(declName, "TLS_ECDHE_") {
+		if strings.HasPrefix(declName, "TLS_RSA_") || strings.HasPrefix(declName, "TLS_ECDHE_") || 
+		strings.HasPrefix(declName, "TLS_AES_") || strings.HasPrefix(declName, "TLS_CHACHA20_") {
```
Comment (pjbgf) [inline staging/src/k8s.io/component-base/cli/flag/ciphersuites_flag_test.go:90]: @liggitt I added the prefix of TLS1.3 ciphers to ensure the test will cover them going forward.

**Comment by pjbgf** [comment]
> hmm, I thought there was a unit test ensuring all supported ciphers were available in this map already… I would have expected a unit test failure if that was not the case

@liggitt that's right, but the test was only covering ciphers that started with `TLS_RSA_` or `TLS_ECDHE`, so they would skip the ciphers on this PR. PTAL

**Comment by liggitt** [comment]
verify is failing because gofmt needs to be run. Run `hack/update-gofmt.sh` and update with the result

**Comment by pjbgf** [comment]
Thank you for pointing that out, @liggitt - just fixed it.

**Comment by liggitt** [comment]
/approve

thanks, go ahead and squash to a single commit, then lgtm

**Pull request merged**